pkgs/applications/networking/browsers/vivaldi/default.nix

```diff
@@ -40,7 +40,18 @@ stdenv.mkDerivation rec {
       (":" + stdenv.lib.makeSearchPathOutput "lib" "lib64" buildInputs)
     + ":$out/opt/vivaldi/lib";
 
+  # Verify that vivaldi-ffmpeg-codecs is built from a compatible Chromium version.
+  preBuild = stdenv.lib.optionalString proprietaryCodecs ''
+    local chromium_version=$(strings opt/vivaldi/vivaldi-bin | grep -A2 -i '^chrome\/' | grep '^[0-9]\+\.[0-9]\+\.[1-9][0-9]\+\.[0-9]\+')
+    if [[ "$chromium_version" != "${vivaldi-ffmpeg-codecs.version}" ]]; then
+      echo "Chromium version mismatch between vivaldi ($chromium_version) and vivaldi-ffmpeg-codecs (${vivaldi-ffmpeg-codecs.version})!"
+      exit 1
+    fi
+  '';
+
   buildPhase = ''
+    runHook preBuild
```

Comment:
Not when overriding buildPhase. [There's a couple packages doing the same thing.](https://github.com/NixOS/nixpkgs/search?utf8=✓&q=runhook+prebuild&type=)